tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -345,34 +345,80 @@ def test_fast_capacity_failover(
-    # test enable fast instance capacity failover
+
+    # Nodes in Single Instance Type CR
+    ice_single_dynamic_nodes = [node for node in dynamic_nodes if "ice-compute-resource" in node]
+    ice_single_static_nodes = [node for node in static_nodes if "ice-compute-resource" in node]
+
+    # Nodes in Multiple Instance Type (overridden with invalid values)
+    ice_multi_dynamic_nodes = [node for node in dynamic_nodes if "ice-cr-multiple" in node]
+    ice_multi_static_nodes = [node for node in static_nodes if "ice-cr-multiple" in node]
+
+    # Nodes in Multiple Instance Type (overridden with values that will trigger an exception)
+    exception_multi_dynamic_nodes = [node for node in dynamic_nodes if "exception-cr-multiple" in node]
+    exception_multi_static_nodes = [node for node in static_nodes if "exception-cr-multiple" in node]
+
+    # Checks Fast Failover in case of Single Instance Type - using RunInstances API
+    # Requires 1 static node in the CR, expects the job to be partially reallocated in a different CR and succeed
     _test_enable_fast_capacity_failover(
         scheduler_commands,
         remote_command_executor,
         clustermgtd_conf_path,
-        static_nodes_in_ice_compute_resource,
-        ice_dynamic_nodes,
+        ice_single_static_nodes,
+        ice_single_dynamic_nodes,
+        target_compute_resource="ice-compute-resource",
+        expected_error_code="InsufficientHostCapacity",
     )
+
+    # Check observability logic
     structured_log_event_utils.assert_that_event_exists(
         cluster, r".+\.slurm_resume_events", "node-launch-failure-count"
     )
     test_cluster_health_metric(["InsufficientCapacityErrors"], cluster.cfn_name, region)
-    # remove logs from slurm_resume log and clustermgtd log in order to check logs after disable fast capacity fail-over
-    remote_command_executor.run_remote_command("sudo truncate -s 0 /var/log/parallelcluster/slurm_resume.log")
-    remote_command_executor.run_remote_command("sudo truncate -s 0 /var/log/parallelcluster/clustermgtd")
-    # test disable ice logic
+
+    # Test behavior with RunInstance when Fast Failover is disabled
+    # Requires 1 static node in the CR, force the job to stay in the CR and expects it to fail
     _test_disable_fast_capacity_failover(
         scheduler_commands,
         remote_command_executor,
         clustermgtd_conf_path,
-        static_nodes_in_ice_compute_resource,
-        ice_dynamic_nodes,
+        ice_single_static_nodes,
+        ice_single_dynamic_nodes,
+        target_compute_resource="ice-compute-resource",
+        expected_error_code="InsufficientHostCapacity",
+    )
+
+    # Checks Fast Failover in case of Multiple Instance Types - using CreateFleet API
+    # Requires 1 static node in the CR, expects the job to be partially reallocated in a different CR and succeed
+    # CreateFleet will return an empty list of instances, that should trigger FFO behavior
+    _test_enable_fast_capacity_failover(
```

Comment:
I was expecting a `_test_enable_fast_capacity_failover` also for the new queue `exception-cr-multiple`, so to test fast failover also for the case when createFleet call generate an exception, why it wasn't considered?

Reply:
Both cases are triggering the same code path. The check that both conditions (exception and empty instance list) trigger the same code path should is tested as Unit Test in the node package.

Repeating this test in the Integration test will add nothing in terms of coverage but will consume many more minutes in terms of execution time.